# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/cs_clip_image.glsl
# Commit: f3b8d237e29a
# Full Hash: f3b8d237e29af262b2042809e13ce6043b75c922
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2021-04-10 03:36:58
# Regressor Bug: 1682195
# File Overlap Count: 1
# Description:
#   Bug 1682195 - Ensure that area outside image mask is cleared if it is first clip. r=gw
#   
#   This fixes a previous bug in the image mask code that the tile rect caused cs_clip_image
#   to use discard on a first clip, the outside area would be left to whatever the initial value
#   of the texture from texture cache was. Since we need to support trimming down to the tile
# ==============================================================================

diff -r b641b7f90108 -r f3b8d237e29a gfx/wr/webrender/res/cs_clip_image.glsl
--- a/gfx/wr/webrender/res/cs_clip_image.glsl	Fri Apr 09 22:55:33 2021 +0000
+++ b/gfx/wr/webrender/res/cs_clip_image.glsl	Fri Apr 09 22:55:34 2021 +0000
@@ -101,7 +101,7 @@
     float alpha = init_transform_fs(vLocalPos);
     vec2 source_uv = clamp(vClipMaskImageUv, vClipMaskUvInnerRect.xy, vClipMaskUvInnerRect.zw);
     float clip_alpha = texture(sColor0, source_uv).r; //careful: texture has type A8
-    oFragColor = vec4(alpha * clip_alpha, 1.0, 1.0, 1.0);
+    oFragColor = vec4(mix(1.0, clip_alpha, alpha), 0.0, 0.0, 1.0);
 }
 
 #ifdef SWGL_DRAW_SPAN